spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/spanner/TransactionalTests.java

```diff
@@ -0,0 +1,111 @@
+package com.google.cloud.spring.autoconfigure.spanner;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.google.cloud.spanner.AbortedException;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spring.autoconfigure.core.GcpContextAutoConfiguration;
+import com.google.cloud.spring.data.spanner.repository.SpannerRepository;
+import io.grpc.Status;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.annotation.Id;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertNotNull;
+
+
+public class TransactionalTests {
```

Comment:
Is it possible to put the tests inside the Spanner module instead of autoconfig?

Reply:
No because we need the autoconfigured dependencies to be the same as in a real application, and referencing autoconfiguration module from spanner module would result in a circular dependency. Source: tried that first :D